client/core/core.go

```diff
@@ -5098,112 +5114,191 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 		}
 		redemptionReserves, err = accountRedeemer.ReserveNRedemptions(redemptionRefundLots, wallets.toAsset)
 		if err != nil {
-			return nil, nil, codedError(walletErr, fmt.Errorf("ReserveNRedemptions error: %w", err))
+			return nil, codedError(walletErr, fmt.Errorf("ReserveNRedemptions error: %w", err))
 		}
 		msgTrade.RedeemSig = &msgjson.RedeemSig{
 			PubKey: pubKeys[0],
 			Sig:    sigs[0],
 		}
-		defer func() {
-			if !success {
-				accountRedeemer.UnlockRedemptionReserves(redemptionReserves)
-			}
-		}()
+		if !form.Queue {
+			defer func() {
+				if !success {
+					accountRedeemer.UnlockRedemptionReserves(redemptionReserves)
+				}
+			}()
+		}
 	}
 
 	// If the from asset is an AccountLocker, we need to lock up refund funds.
 	var refundReserves uint64
 	if isAccountRefund {
 		refundReserves, err = accountRefunder.ReserveNRefunds(redemptionRefundLots, wallets.fromAsset)
 		if err != nil {
-			return nil, nil, codedError(walletErr, fmt.Errorf("ReserveNRefunds error: %w", err))
+			return nil, codedError(walletErr, fmt.Errorf("ReserveNRefunds error: %w", err))
+		}
+		if !form.Queue {
+			defer func() {
+				if !success {
+					accountRefunder.UnlockRefundReserves(refundReserves)
+				}
+			}()
 		}
-		defer func() {
-			if !success {
-				accountRefunder.UnlockRefundReserves(refundReserves)
-			}
-		}()
 	}
 
 	// A non-nil changeID indicates that this is an account based coin. The
 	// first coin is an address and the entire serialized message needs to
 	// be signed with that address's private key.
 	if changeID != nil {
 		if _, msgTrade.Coins[0].Sigs, err = fromWallet.SignMessage(nil, msgOrder.Serialize()); err != nil {
-			return nil, nil, fmt.Errorf("wallet %v failed to sign for redeem: %w", wallets.fromAsset.ID, err)
+			return nil, fmt.Errorf("wallet %v failed to sign for redeem: %w", wallets.fromAsset.ID, err)
 		}
 	}
 
 	commitSig := make(chan struct{})
-	defer close(commitSig) // signals on both success and failure, unlike syncOrderPlaced/piSyncers
+	if !form.Queue {
+		defer close(commitSig) // signals on both success and failure, unlike syncOrderPlaced/piSyncers
+	}
 	c.sentCommitsMtx.Lock()
 	c.sentCommits[prefix.Commit] = commitSig
 	c.sentCommitsMtx.Unlock()
 
+	po := &processOrder{
+		wallets:             wallets,
+		mktID:               mktID,
+		route:               route,
+		coins:               coins,
+		preImg:              preImg,
+		changeID:            changeID,
+		msgOrder:            msgOrder,
+		recoveryCoin:        recoveryCoin,
+		isAccountRedemption: isAccountRedemption,
+		redemptionReserves:  redemptionReserves,
+		isAccountRefund:     isAccountRefund,
+		refundReserves:      refundReserves,
+	}
+
+	if form.Queue {
+		meta := &db.OrderMetaData{
+			Status:     order.OrderStatusPending,
+			Host:       form.Host,
+			ChangeCoin: changeID,
+		}
+
+		corder := coreOrderFromTrade(ord, meta, true)
```

Comment:
Having `coreOrderFromTrade` assign a random temp ID isn't desirable.  `coreOrderFromTrade` is intended to be a transform function and should be deterministic.  If it's called again elsewhere in the code, the ID is different.

I need to consider more carefully, but I believe the temp ID needs to be stored with the `OrderMetaData`